src/Sparrow/Json/Parsing/UnmanagedJsonParserHelper.cs

```diff
@@ -105,6 +105,34 @@ public static void ReadObject(BlittableJsonDocumentBuilder builder, PeepingTomSt
             builder.FinalizeDocument();
         }
 
+        public static void ReadProperty(BlittableJsonDocumentBuilder builder, PeepingTomStream peepingTomStream, UnmanagedJsonParser parser, JsonOperationContext.MemoryBuffer buffer)
+        {
+            builder.ReadProperty();
+            while (builder.Read() == false)
+            {
+                var read = peepingTomStream.Read(buffer.Memory.Span);
+                if (read == 0)
+                    throw new EndOfStreamException("Stream ended without reaching end of json content" + GetPeepingTomBufferAsString(peepingTomStream));
+
+                parser.SetBuffer(buffer, 0, read);
+            }
+            builder.FinalizeDocument();
+        }
+
+        public static async Task ReadPropertyAsync(BlittableJsonDocumentBuilder builder, PeepingTomStream peepingTomStream, UnmanagedJsonParser parser, JsonOperationContext.MemoryBuffer buffer)
+        {
+            builder.ReadProperty();
+            while (builder.Read() == false)
+            {
+                var read = await peepingTomStream.ReadAsync(buffer.Memory).ConfigureAwait(false);
+                if (read == 0)
+                    throw new EndOfStreamException("Stream ended without reaching end of json content" + GetPeepingTomBufferAsString(peepingTomStream));
+
+                parser.SetBuffer(buffer, 0, read);
+            }
+            builder.FinalizeDocument();
```

Comment:
why we need this? Not sure that I follow what are you doing here?

Reply:
we need to treat the property `"@metadata":{}` as a document, to have it as `{"@metadata":{}}` .